src/protocols/secure_channel/CheckinMessage.h

```diff
@@ -100,6 +100,8 @@ class DLL_EXPORT CheckinMessage
         CHIP_CRYPTO_AEAD_NONCE_LENGTH_BYTES + sizeof(CounterType) + CHIP_CRYPTO_AEAD_MIC_LENGTH_BYTES;
 
 private:
+    friend class TestCheckInMsg;
```

Comment:
Please do not friend to test. Which are the methods that you tried to test which were not available?

Reply:
`GenerateCheckInMessageNonce` is one of the functions being tested and is private within the CheckInMessage class.